broker.go

```diff
@@ -1088,11 +1058,12 @@ func (c *offsetCoordinator) commit(
 
 		// get a copy of our connection with the lock, this might establish a new
 		// connection so can take a bit
-		conn, err := c.getConnection()
+		conn, err := c.broker.coordinatorConnection(c.conf.ConsumerGroup)
```

Comment:
I think I agree with this logic. I worry slightly that this could still exhaust an ip's kafka-side connection limit but I agree that this is probably overall better.